### PR TITLE
Allow ElementSubdomainModifer to create complement to moving boundary

### PR DESCRIPTION
**Pull request by jain651**
ref #21383

### CODE REVIEW

File: framework/doc/content/source/transfers/MultiAppMeshFunctionTransfer.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # MultiAppMeshFunctionTransfer
 
-Performs a transfer of field data to/from postprocessor data within sub-applications at locations
-specified by the sub-application position. The transfer utilizes the finite element function of
-the parent application, via a `libMesh::MeshFunction` object, to perform the transfer.
```
Comment (lynnmunday) [inline framework/doc/content/source/transfers/MultiAppMeshFunctionTransfer.md:5]: don't delete this.

Comment (jain651) [reply framework/doc/content/source/transfers/MultiAppMeshFunctionTransfer.md]: This is not the part of this PR. I created this branch from 'devel' branch. Now, I have re based it from 'next'. Now this PR has only my changes.

**Comment by jain651** [comment]
Hi @dschwen Would you please help me to resolve the issue? I believe that the 'Test debug' is a good starting point as it fails even on single processor.

**Comment by GiudGiud** [comment]
Looks like a BC is accessing an array out of range. Likely this is either at:
- a new side that was added when the boundary moved
- an old side that should have been deleted when the boundary moved

File: framework/src/userobject/ElementSubdomainModifier.C [hunk 2]
```diff
@@ -189,12 +217,22 @@ ElementSubdomainModifier::updateBoundaryInfo(MooseMesh & mesh,
     for (auto side : elem->side_index_range())
     {
       const Elem * neighbor = elem->neighbor_ptr(side);
+      if (_complement_moving_boundary_specified)
```
Comment (bwspenc) [inline framework/src/userobject/ElementSubdomainModifier.C:220]: This code seems like it should go down in the block starting in line 238, where a similar operation is done on the moving boundary (and where's some extra code for parallel cases).

**Review by bwspenc — CHANGES_REQUESTED** [review]

**Comment by dschwen** [comment]
Replaced by #22675

**Pull request closed without merging**